code/modules/atmospherics/machinery/components/fusion/hfr_parts.dm

```diff
@@ -392,3 +392,76 @@
 	-In case of a power shortage, the fusion reaction will CONTINUE but the cooling will STOP<BR><BR>\
 	The writer of the quick guide will not be held responsible for misuses and meltdown caused by the use of the guide, \
 	use more advanced guides to understando how the various gases will act as moderators."
+
+/obj/item/hfr_box
+	name = "HFR box"
+	desc = "If you see this, call the police."
+	icon = 'icons/obj/atmospherics/components/hypertorus.dmi'
+	icon_state = "box"
+	///What kind of box are we handling?
+	var/box_type = "impossible"
+
+/obj/item/hfr_box/corner
+	name = "HFR box corner"
+	desc = "Place this as the corner of your 3x3 multiblock fusion reactor"
+	icon_state = "box_corner"
+	box_type = "corner"
+
+/obj/item/hfr_box/body
+	name = "HFR box body"
+	desc = "Place this on the sides of the core box of your 3x3 multiblock fusion reactor"
+	icon_state = "box_body"
+	box_type = "body"
+
+/obj/item/hfr_box/core
+	name = "HFR box core"
+	desc = "Activate this with a multitool to deploy the full machine after setting up the other boxes"
+	icon_state = "box_core"
+	box_type = "core"
+	var/list/body_components = list(
+		/obj/machinery/hypertorus/interface,
+		/obj/machinery/atmospherics/components/unary/hypertorus/fuel_input,
+		/obj/machinery/atmospherics/components/unary/hypertorus/waste_output,
+		/obj/machinery/atmospherics/components/unary/hypertorus/moderator_input
+	)
+
+/obj/item/hfr_box/core/multitool_act(mob/living/user, obj/item/I)
+	. = ..()
+	var/list/parts = list()
+	for(var/obj/item/hfr_box/box in orange(1,src))
+		var/direction = get_dir(src, box)
+		if(box.box_type == "corner")
+			if(ISDIAGONALDIR(direction))
+				box.dir = direction
+				parts |= box
+			continue
+		if(box.box_type == "body")
+			if(direction in GLOB.cardinals)
+				box.dir = DIRFLIP(direction)
+				parts |= box
+			continue
+	if(parts.len == 8)
+		build_reactor(parts)
+	return
+
+/obj/item/hfr_box/core/proc/build_reactor(list/parts)
+	for(var/obj/item/hfr_box/box in parts)
+		if(box.box_type == "corner")
+			var/obj/machinery/hypertorus/corner/corner = new(box.loc)
```

Comment:
Why not set box_type to the actual typepath you want it to create, and then use that with new() to do the initialization?

Reply:
That would work only for the corners since those boxes only spawn that type

The body boxes can be four different types. The only way that would work is to make four different body boxes

Reply:
I could do that now that I think about it, just so people can spawn the hfr and is not just random

Reply:
should be done in the next commit